pkg/sources/github/github.go

```diff
@@ -470,31 +470,36 @@ func (s *Source) scan(ctx context.Context, installationClient *github.Client, ch
 	s.repos = reposToScan
 
 	var scanErrs []error
-	for i, repoURL := range s.repos {
-		i, repoURL := i, repoURL
+	repoCnt := len(s.repos)
+	for _, repoURL := range s.repos {
+		repoURL := repoURL
 		s.jobPool.Go(func() error {
 			if common.IsDone(ctx) {
 				return nil
 			}
+			cnt := s.Counter.IncTotal()
 
-			// TODO: set progress complete is being called concurrently with i
-			s.setProgressCompleteWithRepo(i, progressIndexOffset, repoURL)
+			defer func() {
+				s.Counter.IncSuccess()
+			}()
+			s.setProgressCompleteWithRepo(progressIndexOffset, repoURL)
 			// Ensure the repo is removed from the resume info after being scanned.
 			defer func(s *Source, repoURL string) {
 				s.resumeInfoMutex.Lock()
 				defer s.resumeInfoMutex.Unlock()
 				s.resumeInfoSlice = sources.RemoveRepoFromResumeInfo(s.resumeInfoSlice, repoURL)
 			}(s, repoURL)
 
+			var err error
 			if !strings.HasSuffix(repoURL, ".git") {
-				scanErrs = append(scanErrs, fmt.Errorf("repo %s does not end in .git", repoURL))
+				err = fmt.Errorf("repo %s does not end in .git", repoURL)
+				scanErrs = append(scanErrs, err)
 				return nil
 			}
```

Comment:
**Nit (optional):** The assignment to `err` here isn't needed since we `return nil`.